test/test_data/glibc.py

```diff
@@ -0,0 +1,11 @@
+mapping_test_data = [
+    {"product": "Glibc", "version": "2.3.4", "version_strings": ["GLIBC_2.3.4"]}
```

Comment:
Here, your  version regex is `r"GLIBC ([0-9]+\.[0-9]+\.[0-9]+)"` while your version_strings has `["GLIBC_2.3.4"]` which doesn't satisfy your given regex. Always check your regex on a sample string. Here is the online site for that: https://regex101.com/. You can check regex locally on Python but this site would be helpful if you are creating a hard regex because it provides quick reference and explanation of your regex.

You would also have to add a string satisfying contains string regex in mapping_test_data's version_strings list. You can also have multiple version strings. So, I recommend adding regex of contains string in version_string also because that can also give us valid version. As a rule of thumb, whatever you add in version_string can be added in contains_string but reverse isn't true since you can also add strings that don't have version string but can be only found in the product you are making checker for as contains strings.

> One can say that, CONTAINS_STRINGS are superset of VERSION_STRINGS.

Reply:
Okay. Thanks!
I will look into that